src/components/presentational/DataChart/DataChart.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useContext } from 'react'
+import { add, format, isToday } from 'date-fns'
+import { CardTitle, LayoutContext, LoadingIndicator } from '../../presentational'
+import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./DataChart.css"
+import { AxisDomain } from 'recharts/types/util/types'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+
+export interface DataChartProps {
+    title?: string
+    intervalType?: "Week" | "Month",
+    intervalStart: Date,
+    data: any[],
+    dataKeys?: string[],
+    hasAnyData: boolean,
+    tooltip: ({ active, payload, label }: any) => React.JSX.Element | null,
+    chartType: "Line" | "Bar" | "Area"
+    options?: LineChartOptions | BarChartOptions | AreaChartOptions
+    hideIfNoData?: boolean
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface LineChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    domainMin?: number | "Auto"
+}
+
+export interface BarChartOptions {
+    barColor?: ColorDefinition
+    thresholds?: BarChartThreshold[]
+}
+
+export interface BarChartThreshold {
+    value: number
+    referenceLineColor?: ColorDefinition
+    overThresholdBarColor?: ColorDefinition
+}
+
+export interface AreaChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    areaColor?: ColorDefinition | ColorDefinition[],
+}
+
+export default function DataChart(props: DataChartProps) {
+    let layoutContext = useContext(LayoutContext);
+    let intervalType = props.intervalType || "Month";
+
+    let chartHasData = props.data.length > 0;
+
+    const DayTick = ({ x, y, stroke, payload }: any) => {
```

Comment:
So this chart is still definitely time-based and just supports week or month formatting and daily values.  Unfortunately "DailyDataChart" was taken (I guess I should have named it DailyDeviceDataChart).

How about "TimeSeriesChart" - that actually feels like it is the distinguishing factor here.  We have time-series data and want to graph it...the thing that separates this from any other data chart is that we have written some logic for the x-axis etc...to properly display day of week / month etc...

Reply:
I think we could also take a swing at extending this into Intraday data with https://github.com/CareEvolution/MyDataHelpsUI/pull/247/files - @reneefromhold I am not sure we should have a "LineChart" component...we could try to use this one and modify it so the x-axis properly supports intraday data (I think seems straightforward - just intervalType = day)

Reply:
Yeah - @reneefromhold have spoken; mainly she asked if I could integrate the line threshold colors she implemented there.  I'm working on updating things so that she could remove her LineChart in favor of this one.